literal/decimal_util.go

```diff
@@ -0,0 +1,73 @@
+package literal
+
+import (
+	"fmt"
+	"math/big"
+	"strings"
+)
+
+func decimalStringToBytes(decimalStr string) ([16]byte, int32, int32, error) {
```

Comment:
doc plz

Reply:
also, I suggest this function starts with checking the pattern. e.g. the only valid pattern is
`[+-][\d]{0,38}.[\d]{0,38}` or something like that.